compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -998,6 +998,17 @@ impl<'tcx> LocalDecl<'tcx> {
         )
     }
 
+    pub fn has_pattern_match(&self) -> bool {
+        matches!(
+            self.local_info(),
+            LocalInfo::User(BindingForm::Var(VarBindingForm {
+                binding_mode: ty::BindingMode::BindByValue(Mutability::Not),
```

Comment:
This seems more restrictive than the method name would imply. Can we check the mutability in the caller instead to make this more generic like the name?

Reply:
yeah, it's not a general api.
I found the caller already has `decl.can_be_made_mutable()` to make it safe to add `mut`, so we don't need to check it.